vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogiteminitialization.rb

```diff
@@ -1,126 +1,220 @@
 
-  tags_hash = {}
+    new_service_name = "#{@service.name}-#{Time.now.strftime('%Y%m%d-%H%M%S')}" if new_service_name.blank?
+    log(:info, "Service name: #{new_service_name}")
+    @service.name = new_service_name
 
-  # Loop through all of the tags and build an options_hash from them
-  dialog_options.each do |k, v|
-    if tags_regex =~ k
-      # Convert key to symbol
-      tag_category = Regexp.last_match[1].to_sym
-      tag_value = v.downcase
+    return if new_service_description.blank?
+    log(:info, "Service description #{new_service_description}")
+    @service.description = new_service_description
+    log(:info, "Processing service_naming...Complete", true)
+  end
 
-      unless tag_value.blank?
-        $evm.log("info", "Adding category:<#{tag_category.inspect}> tag:<#{tag_value.inspect}> to tags_hash")
-        tags_hash[tag_category] = tag_value
+  # service_tagging - tag the service with tags in dialogs_tags_hash[0]
+  def service_tagging(merged_tags_hash)
+    log(:info, "Processing service_tagging...", true)
+    # Look for tags with a sequence_id of 0 to tag the service
+    merged_tags_hash.each do |key, value|
+      log(:info, "Processing Tag Key: #{key.inspect}  value: #{value.inspect}")
+      next if value.blank?
+      tag_category = key.downcase
+      if value.kind_of?(Array)
+        value.each do |tag_entry|
+          tag_value = tag_entry.downcase
+          $evm.log(:info, "Adding tag category: #{tag_category} tag: #{tag_value} to Service: #{@service.name}")
+          @service.tag_assign("#{tag_category}/#{tag_value}")
+        end
+      else
+        tag_value = value.downcase
+        log(:info, "Adding Tag: {#{key.inspect} => #{value.inspect}} to Service: #{@service.name}")
+        @service.tag_assign("#{tag_category}/#{tag_value}")
       end
     end
   end
-  $evm.log("info", "Inspecting tags_hash:<#{tags_hash.inspect}>")
-  tags_hash
-end
 
-# Look for service dialog variables in the dialog options hash that start with "option_[0-9]",
-def get_options_hash(dialog_options)
-  # Setup regular expression for service dialog tags
-  options_regex = /^dialog_option_\d*_(.*)/
-  options_hash = {}
-
-  # Loop through all of the options and build an options_hash from them
-  dialog_options.each do |k, v|
-    if options_regex =~ k
-      option_key = Regexp.last_match[1].to_sym
-      option_value = v
-
-      unless option_value.blank?
-        $evm.log("info", "Adding option_key:<#{option_key.inspect}> option_value:<#{option_value.inspect}> to options_hash")
-        options_hash[option_key] = option_value
-      end
+  # use this to name the vm from matching_options_hash[:vm_name] || matching_options_hash[:vm_target_name]
+  def get_vm_name(merged_options_hash, prov)
+    log(:info, "Processing get_vm_name", true)
+    new_vm_name = merged_options_hash[:vm_name] || merged_options_hash[:vm_target_name] rescue nil
+
+    if new_vm_name.blank?
```

Comment:
@tinaafitz 
can we use
new_vm_name = prov.get_option(:vm_target_name) if new_vm_name.blank?
and then just use the else clause block